lib/stream-utils.ts

```diff
@@ -1,5 +1,9 @@
+import * as crypto from "crypto";
 import { Readable } from "stream";
 
+const HASH_ALGORITHM = "sha256"; // TODO algorithm?
```

Comment:
the algorithm used here needs to match the one used by the key-binary lookup service, but I would rather both of them to be ```sha1``` instead of ```sha256```` probably

Reply:
let's do it then

Reply:
we will, when it's the time and when the PR will be around it. now is neither the time nor the PR. I'm looking to ship incremental value here, not solve everything :)